### PR TITLE
[tests-only][full-ci]add test for copy and move multiple files at once

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
In this PR, steps are implemented for copying and moving multiple resources at once using different methods: `dropdown-menu, keyboard, batch-action, drag-drop, drag-drop-breadcrumb`.

## Related Issue

- https://github.com/owncloud/ocis/issues/7363

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] e2e tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/e2e/cucumber/features/smoke/copyMove.feature [hunk 1]
```diff
@@ -56,4 +66,40 @@ Feature: Copy
...
       | resource | to          |
       | Sub2     | PARENTCopy4 |
+
+    And "Alice" opens the "files" app
+    And "Alice" opens folder "PARENT"
+    And "Alice" copies the following resources to "PARENT/Sub1" at once using dropdown-menu
+      | resource        |
+      | fileToCopy1.txt |
+      | fileToCopy2.txt |
+    And "Alice" copies the following resources to "PARENT/Sub1" at once using batch-action
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/smoke/copyMove.feature:76]: is it different from step `copies the following resource using batch-action`

File: tests/e2e/cucumber/features/smoke/copyMove.feature [hunk 2]
```diff
@@ -56,4 +66,40 @@ Feature: Copy
     And "Alice" moves the following resource using drag-drop-breadcrumb
       | resource | to          |
       | Sub2     | PARENTCopy4 |
+
+    And "Alice" opens the "files" app
+    And "Alice" opens folder "PARENT"
+    And "Alice" copies the following resources to "PARENT/Sub1" at once using dropdown-menu
+      | resource        |
+      | fileToCopy1.txt |
+      | fileToCopy2.txt |
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/smoke/copyMove.feature:75]: can we do it with more resources? doing copy/move between folders that would reduce number of files to create

File: tests/e2e/cucumber/steps/ui/resources.ts [hunk 3]
```diff
@@ -128,6 +128,30 @@ When(
...
+    stepUser: string,
+    actionType: string,
+    newLocation: string,
+    method: string,
+    stepTable: DataTable
+  ): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const resourceObject = new objects.applicationFiles.Resource({ page })
+
+    const resources = [].concat(...stepTable.rows())
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/ui/resources.ts:144]:
```suggestion
    const resources = stepTable.rows()
```

File: tests/e2e/cucumber/features/smoke/copyMove.feature (hunk 1 above)
Comment (PrajwolAmatya) [reply tests/e2e/cucumber/features/smoke/copyMove.feature:76]: yeah, the step above is for copying/moving single resource whereas, the new step is for copying/moving multiple resource at once using the copy button from batch action bar.
image

File: tests/e2e/cucumber/steps/ui/resources.ts (hunk 3 above)
Comment (PrajwolAmatya) [reply tests/e2e/cucumber/steps/ui/resources.ts:144]: `stepTable.rows()` will return two-dimensional array. For easier processing, converted the array into one-dimensional array.

File: tests/e2e/cucumber/features/smoke/copyMove.feature (hunk 2 above)
Comment (ScharfViktor) [reply tests/e2e/cucumber/features/smoke/copyMove.feature:75]: Yes, I agree. You are moving 2 files at a time. It is pretty few for checking
How said Sajan, I suggest to move all files `fileToCopyN.txt` to folder sub1, after that from sub1 to sub2 ...

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 4]
```diff
@@ -104,6 +104,7 @@ const onlyOfficeCanvasCursorSelector = '#id_target_cursor'
 const collaboraCanvasEditorSelector = '.leaflet-layer'
 const textEditorTextArea = '#text-editor-input'
 const filesContextMenuAction = 'div[id^="context-menu-drop"] button.oc-files-actions-%s-trigger'
+const highlightedTableRowSelector = '//tbody/tr[contains(@class, "oc-table-highlighted")]'
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:107]:
```suggestion
const highlightedTableRowSelector = '#files-space-table tr.oc-table-highlighted'
```

File: tests/e2e/cucumber/features/smoke/copyMove.feature [hunk 5]
```diff
@@ -56,4 +66,68 @@ Feature: Copy
...
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/smoke/copyMove.feature:75]: can we also include some folders?

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 6]
```diff
@@ -104,6 +104,7 @@ const onlyOfficeCanvasCursorSelector = '#id_target_cursor'
 const collaboraCanvasEditorSelector = '.leaflet-layer'
 const textEditorTextArea = '#text-editor-input'
 const filesContextMenuAction = 'div[id^="context-menu-drop"] button.oc-files-actions-%s-trigger'
+const highlightedTableRowSelector = '#files-space-table tr.oc-table-highlighted'
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:107]:
```suggestion
const highlightedFileRowSelector = '#files-space-table tr.oc-table-highlighted'
```

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 7]
```diff
@@ -684,6 +689,110 @@ export const pasteResource = async (
...
+      break
+    }
+    case 'drag-drop': {
+      const source = page.locator(highlightedTableRowSelector).first()
+      const target = page.locator(util.format(resourceNameSelector, newLocation))
+
+      await Promise.all([...waitForMoveResponses, source.dragTo(target)])
+
+      await target.click()
+      break
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:778]: we don't need `waitForResources` here?

File: tests/e2e/support/objects/app-files/resource/actions.ts [hunk 8]
```diff
@@ -684,6 +689,114 @@ export const pasteResource = async (
...
+      await Promise.all([...waitForMoveResponses, source.dragTo(target)])
+
+      await target.click()
+      await waitForResources({
+        page,
+        names: resources
+      })
+      break
+    }
+  }
```
Comment (saw-jan) [inline tests/e2e/support/objects/app-files/resource/actions.ts:797]: Then now we can wait for resources at the last
```suggestion
    }
    await waitForResources({
        page,
        names: resources
      })
  }
```

**Review by saw-jan — COMMENTED** [review]
LGTM :+1: :sparkles:

**Review by ScharfViktor — APPROVED** [review]

**Pull request merged**